connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala

```diff
@@ -189,9 +245,54 @@ object SparkConnectClient {
     }
 
     def build(): SparkConnectClient = {
-      val channelBuilder = ManagedChannelBuilder.forAddress(host, port).usePlaintext()
-      val channel: ManagedChannel = channelBuilder.build()
+      def insecureCredentials(): ChannelCredentials = {
+        InsecureChannelCredentials.create()
+      }
+      val creds = isSslEnabled match {
+        case None => insecureCredentials()
```

Comment:
NIT, you can combine cases, e.g.: `case None | Some(false) => insecureCredentials()`